### PR TITLE
add opt-in for identifiable random generation.

### PR DESCRIPTION
**Pull request by simone-licciardi**
- now control flow of the sampler can be explicitly controlled from config.
- control flow of sampler is now independent on how the Scheduler was generated
- we kept for backward-compatibility the possibility of allowing to identify the flow that generated a batch.
[IMPORTANT] Code that relied on this faeature needs to adjust their config file to continue working. Code that didn't rely on this feature need not.

### CODE REVIEW

**Comment by simone-licciardi** [comment]
Closes #248

**Comment by simone-licciardi** [comment]
@fbanelli can you check whether this solution is still compatible with your code (see description of the commit)?
Moreover, if you want to make further comments, feel free :)

File: config/main.yaml [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-batches_per_flow_batch: 1 # Number of batches of (imgs1, imgs2, flows) tuples per flow batch
+batch_size: 1 # Number of (img1, img2, flow) tuples per data batch
+flow_fields_per_batch: 1 # Number of unique flow fields per data batch
+batches_per_flow_batch: 1 # Number of consecutive batches of (imgs1, imgs2, flows) generated with each set of flow fields
 randomize: false # Whether to randomize the order of the files
 loop: true # Whether to loop through the files indefinitely
-device_ids: [0]
+identifiable_flow: false # Whether to allow identification of the flow generating that image
+device_ids: [0] # Indexes of the machine's visible device list that are selected.
+                # Omitt to select all avalable devices.
```
Comment (fbanelli) [inline config/main.yaml:10]: typo: omit

File: src/synthpix/sampler/synthetic.py [hunk 2]
```diff
@@ -104,6 +105,9 @@ def __init__(
             histogram: Optional histogram to match during image generation.
                 Should be a jnp.ndarray of shape (256,).
                 NOTE: Histogram equalization is very slow!
+            identifiable_flow: Optional, makes sampled data batches' flows
+                identifiable for the downstream user.
+                NOTE: identifiable_flow is slow!
```
Comment (fbanelli) [inline src/synthpix/sampler/synthetic.py:110]: I see that in #251 you have added a benchmark. Please quantify how much slower this is (on our workstation)

**Review by fbanelli — CHANGES_REQUESTED** [review]
minor comments

File: config/main.yaml (hunk 1 above)
Comment (antonioterpin) [reply config/main.yaml:10]: See comment in #251: these comments are apparently unmantainable

**Review by antonioterpin — CHANGES_REQUESTED** [review]
In which order should #249 and #251 be considered? Let's only do #251, seems like it is more complete and close this one.
